lib/AST/ASTPrinter.cpp

```diff
@@ -5755,9 +5754,28 @@ swift::getInheritedForPrinting(
           isa<EnumDecl>(decl) &&
           cast<EnumDecl>(decl)->hasRawType())
         continue;
-      Results.push_back({TypeLoc::withoutLoc(proto->getDeclaredInterfaceType()),
-                      /*isUnchecked=*/false});
+      protocols.insert(proto);
+    }
+  }
+
+  for (size_t i = 0; i < protocols.size(); i++) {
+    auto proto = protocols[i];
+
+    if (!options.shouldPrint(proto)) {
+      // If private stdlib protocols are skipped and this is a private stdlib
+      // protocol, see if any of its inherited protocols are public. Those
+      // protocols can affect the user-visible behavior of the declaration, and
+      // should be printed.
+      if (options.SkipPrivateStdlibDecls &&
+          proto->isPrivateStdlibDecl(!options.SkipUnderscoredStdlibProtocols)) {
+        auto inheritedProtocols = proto->getInheritedProtocols();
+        protocols.insert(inheritedProtocols.begin(), inheritedProtocols.end());
```

Comment:
Is it possible that one of these protocols would be hidden?

Reply:
Yes, in that case we'll just skip it on L5764. This actually happens with the `NSError` types: they are imported with a conformance to `_BridgedStoredNSError` (hidden), which inherits from `_ObjectiveCBridgeableError` (also hidden), so this logic will look into the parent protocol of `_ObjectiveCBridgeableError`, which is `Error`, and the conformance to `Error` will be printed.